src/inmanta/plugins.py

```diff
@@ -852,24 +946,15 @@ def __call__(self, *args: object, **kwargs: object) -> object:
             warnings.warn(PluginDeprecationWarning(msg))
         self.check_requirements()
 
-        def new_arg(arg: object) -> object:
-            if isinstance(arg, Context):
-                return arg
-            elif isinstance(arg, Unknown) and self.is_accept_unknowns():
-                return arg
-            else:
-                return DynamicProxy.return_value(arg)
-
-        new_args = [new_arg(arg) for arg in args]
-        new_kwargs = {k: new_arg(v) for k, v in kwargs.items()}
```

Comment:
Why is it safe to remove this?

Reply:
the jinja integration also uses it

https://github.com/inmanta/std/blob/master/plugins/__init__.py#L175

I'll put it back

Reply:
I clarified that one: it get the pre-processed args from the higher level